### DISCUSSION TITLE
Pop-up menu on long-press of modifier keys

### DISCUSSION THREAD

**Discussion by thomasaarholt** [helix-editor/helix#11491] [General]
I am very fond of the menus that pop-up to show the keymap with a brief explanation when entering minor modes, like `g` and `space`. 
I find it very hard to remember what it is that the modifier keys do, like `Ctrl-i` to jump forward on the jumplist. 
Could it be an idea to (optionally) show a pop-up menu when the user long-presses Ctrl, Alt, or Shift? (Shift is rather unused, so maybe not that one.)

<img width="500" alt="Screenshot 2024-08-14 at 21 22 38" src="https://github.com/user-attachments/assets/1b38d1c1-e26e-4f41-95e6-883ecc3ebd50">

**Comment by pascalkuthe** [comment]
I don't thinkwe can detect that kn a terminal. We only get a single keypress event.

**Comment by thomasaarholt** [comment]
I see. How about double-tapping the modifier key, within some short period of time? Or would that interfere with other usage?

**Comment by pascalkuthe** [comment]
That is also not available on a terminal. Modifiers dint have their own keypress events they ar only flags applied to other key events
